Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt

```diff
@@ -56,45 +56,41 @@ class VaccinationStorage @Inject constructor(
                 Timber.tag(TAG).v("Removing data for %s", it)
                 remove(it)
             }
-            persons.forEach {
-                val raw = gson.toJson(it)
-                val identifier = it.identifier
-                Timber.tag(TAG).v("Storing vaccinatedPerson %s -> %s", identifier, raw)
-                putString("$PKEY_PERSON_PREFIX${identifier.groupingKey}", raw)
+            persons.groupDataByIdentifier().forEach {
+                if (it.vaccinations.isNotEmpty()) {
+                    val raw = gson.toJson(it)
+                    val identifier = it.identifier
+                    Timber.tag(TAG).v("Storing vaccinatedPerson %s -> %s", identifier, raw)
+                    putString("$PKEY_PERSON_PREFIX${identifier.groupingKey}", raw)
+                }
             }
         }
     }
 
-    /*
-    * Groups vaccinations by identifier
-    * Performs data migration if identifier changes
-    */
-    suspend fun reorganizeData() {
-        save(load().groupDataByIdentifier())
-    }
-
     companion object {
         private const val TAG = "VaccinationStorage"
         private const val PKEY_PERSON_PREFIX = "vaccination.person."
     }
 }
 
 internal fun Set<VaccinatedPersonData>.groupDataByIdentifier(): Set<VaccinatedPersonData> {
```

Comment:
for this fun the following **suggestion**:

internal fun Set<VaccinatedPersonData>.groupDataByIdentifier(): Set<VaccinatedPersonData> =
    filterNot { it.vaccinations.isNullOrEmpty() }
        .groupBy { it.identifier }
        .mapNotNull { entry ->
            val personDataList = entry.value
            if (personDataList.isEmpty()) {
                Timber.v("Person data list was empty, returning early")
                return@mapNotNull null
            }

            val newestData = personDataList.maxByOrNull {
                it.lastBoosterNotifiedAt ?: Instant.EPOCH
            }
            val vaccinations = personDataList.flatMap { it.vaccinations }.toSet()
            newestData?.copy(vaccinations = vaccinations)
        }.toSet()